types/canonical.go

```diff
@@ -33,6 +34,25 @@ func CanonicalizeBlockID(bid cmtproto.BlockID) *cmtproto.CanonicalBlockID {
 	return cbid
 }
 
+// CanonicalizeBlobID transforms the given BlobID to a CanonicalBlobID.
+func CanonicalizeBlobID(protoBlobID cmtproto.BlobID) *cmtproto.CanonicalBlobID {
+	blobID, err := BlobIDFromProto(&protoBlobID)
+	if err != nil {
+		panic(fmt.Sprintf("error parsing BlobID from protobuf: %v", err))
+	}
+
+	if blobID.IsNil() {
```

Comment:
Looking at how this is done for blocks , I see that there is a check `blobID == nil` . Not sure what the diff is but maybe we want to include the same check here? 
```suggestion
	if blobID == nil || blobID.IsNil() {
```

Reply:
`BlobIDFromProto()` returns a `BlobID` not a `*BlobID`, therefore the `blobID` variable cannot be `nil`. In contrast, `BlockIDFromProto()` returns  a `*BlockID`, therefore the code must check if it's `nil` before calling the `IsNil()` method to prevent a panic.

So in this case we don't need the `nil` check.